src/factory/mongoCDCService.ts

```diff
@@ -73,45 +97,68 @@ export const mongoCDCService = {
         TE.bind("collection", ({ database }) =>
           mongoDBServiceClient.getResource(database, resourceName),
         ),
+        // Checking that the lease collection {leaseResourceName} exists
+        // If not, a default lease container will be created in the next steps
         TE.bind("leaseCollection", ({ database }) =>
-          mongoDBServiceClient.getResource(database, leaseResourceName),
-        ),
-        TE.bind("leaseDocument", ({ leaseCollection }) =>
-          mongoDBServiceClient.getItemByID(
-            leaseCollection,
-            pipe(
-              opts,
-              O.fromNullable,
-              O.chainNullableK((options) => options.prefix),
-              O.toUndefined,
+          pipe(
+            O.fromNullable(leaseResourceName),
+            O.fold(
+              () => TE.right(O.none),
+              (lease) =>
+                pipe(
+                  mongoDBServiceClient.getResource(database, lease),
+                  TE.map((lcontainer) => O.some(lcontainer)),
+                ),
             ),
           ),
         ),
-        TE.chain(({ collection, leaseDocument }) =>
+        // Checking from the lease container {leaseResourceName that there is a continuationToken stored
+        TE.bind("continuationToken", ({ leaseCollection }) =>
           pipe(
-            leaseDocument,
-            O.chain(
-              flow(
-                ContinuationTokenItem.decode,
-                E.mapLeft((errs) =>
-                  Error(errorsToReadableMessages(errs).join("|")),
+            leaseCollection,
+            O.fold(
+              () => TE.right(O.none),
+              (container) =>
+                pipe(
+                  O.fromNullable(opts),
+                  O.chainNullableK((options) => options.prefix),
+                  O.fold(
+                    () => TE.right(O.none),
+                    (id) => mongoDBServiceClient.getItemByID(container, id),
+                  ),
                 ),
-                E.map((leaseToken) => leaseToken.lease),
-                O.fromEither,
-              ),
             ),
-            O.toUndefined,
-            (lease) =>
-              TE.fromEither(
-                watchChangeFeed(
-                  collection as Collection,
-                  processResults,
-                  lease,
+          ),
+        ),
+        TE.chain(
+          ({ collection, continuationToken, leaseCollection, database }) =>
+            pipe(
+              leaseCollection,
+              O.fold(
+                () =>
+                  getOrCreateMongoCollection(
+                    database as Db,
+                    MONGO_LEASE_COLLECTION_NAME,
+                  ),
+                (lContainer) => TE.right(lContainer),
```

Comment:
```suggestion
                TE.right,
```

Reply:
I have a compilation error with  TE.right,

Reply:
It's strange. BTW it is only a cosmetic consideration so we can safely keep it unmodified